### PR TITLE
Allow Verbose as an Option to cvxpy.Problem#solve

### PR DESCRIPTION
**Pull request by pmn4**
Originally, I simply wanted to remove the hard-coded `verbose=True` setting for `Problem#solve` when the `solver` is specified on the `EfficientFrontier` instance, but then thought it to be a valuable option when debugging your code, so I bubbled the option up to all of the optimization methods: min_variance, max_sharpe, etc.

_(fwiw, I am still very new to Python, so please feel free to suggest and code edits!)_

### CODE REVIEW

**Comment by robertmartin8** [comment]
Hi @pmn4 

Thanks for the suggestion! I take your point that it would be nice to change the verbosity, but I'm not sure I want this to be a top-level API. I'm a little more inclined to have it as a "hidden API" like with the solver, so someone can set `ef.verbose = True` directly. Alternatively, it might be better suited as an optional argument in the constructor of the `BaseConvexOptimizer`.

**Comment by pmn4** [comment]
Hi @robertmartin8, I added `verbose` as a property of `BaseOptimizer` then broadly replaced "verbose as a parameter" with uses of the instance variable (don't hesitate to lmk I went too far!).

File: pypfopt/base_optimizer.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def __init__(self, n_assets, tickers=None):
             self.tickers = tickers
         # Outputs
         self.weights = None
+        self.verbose = False
```
Comment (pmn4) [inline pypfopt/base_optimizer.py:50]: I added the instance variable to the constructor but made it _really_ hidden by not including it in the `initialize` signature. If that is _too_ hidden, lmk and I will update to
``` Python
#  def __init__(self, n_assets, tickers=None):
def __init__(self, n_assets, tickers=None, verbose=False):
```

**Comment by robertmartin8** [comment]
@pmn4 

I really appreciate the work you've done so far! I've thought about this a little bit more and I think I'm ready to move `solver` and `verbose` into optional arguments in the constructor for the class. So someone will be able to call:

```python
ef = EfficientFrontier(mu, S, weight_bounds=(-1,1), ..., solver="SCS", verbose=False)

ef = EfficientFrontier(mu, S, verbose=True)
```

So let's move it into an optional parameter for the `BaseConvexOptimizer` constructor. If you are keen, you can also try moving `solver`, but if you just want to fix the verbose parameter that's totally fine.

Looking forward to your commits, will try to get it merged by end of week!

Cheers,
Robert

**Comment by robertmartin8** [comment]
@pmn4 

`.ipynb` files are jupyter notebooks, which are part of the documentation. I suggest you leave the examples/documentation mostly as is, apart from the in-line documentation explaining the verbose parameter. Because none of your commits are backwards incompatible, the previous examples should still be correct.

I've read through the commits, everything looks good except one thing: I don't want to change the signature of `portfolio_performance`. Within `ef.portfolio_performance()`, the `verbose` parameter determines whether or not we print the returns, volatility and sharpe. This is separate to the `verbose` parameter that you have added, which determines whether or not the optimiser results are printed. These should be separate because the vast majority of users will not care about the optimiser results but may care about the verbose portfolio performance. 

```python
# prints portfolio performance but not optimiser logs
ef = EfficientFrontier(mu, S, verbose=False)
ef.max_sharpe()
ef.portfolio_performance(verbose=True)

# only prints optimiser logs
ef = EfficientFrontier(mu, S, verbose=True)
ef.max_sharpe()
results = ef.portfolio_performance(verbose=False)
```

I understand that it's not ideal to reuse the word `verbose`, but I believe that this is the right decision because it preserves backwards-compatibility. 

Cheers,
Robert

File: pypfopt/cla.py [hunk 2]
```diff
@@ -445,15 +445,15 @@ def set_weights(self, _):
         # Overrides parent method since set_weights does nothing.
         raise NotImplementedError("set_weights does nothing for CLA")
 
-    def portfolio_performance(self, verbose=False, risk_free_rate=0.02):
+    def portfolio_performance(self, risk_free_rate=0.02, verbose=False):
```
Comment (pmn4) [inline pypfopt/cla.py:448]: oops, I didn't necessarily mean to reorder these. I suppose it makes sense? idk, is there a Python style guide for this type of thing?

Comment (openerror) [reply pypfopt/cla.py:448]: Robert says in the [docs](https://pyportfolioopt.readthedocs.io/en/latest/Contributing.html) that he prefers Python code in [Black](https://github.com/psf/black) style. I haven't looked at it personally, but maybe you can see if it's relevant to the keyword ordering?

**Comment by robertmartin8** [comment]
I'm not that fussy about the order as long as it is consistent with the
rest of the package

**Pull request merged**